### DISCUSSION TITLE
PerfView showing "Event Microsoft-DotNETCore-SampleProfiler/Thread/Sample" in many places.

### DISCUSSION THREAD

**Discussion by kiranongithub** [microsoft/perfview#1568] [Q&A]
Hi, Please forgive me if I am asking obvious questions. I am trying to learn PerfView basics

I did a `dotnet-trace` on a .NET Core 6.0 application from within a docker container. I then copied the `.nettrace` file to my windows desktop and opened in the latest version(2.0.74) of PerfView.

**Question**: I keep seeing details of "Event Microsoft-DotNETCore-SampleProfiler/Thread/Sample" in By Name view, Call Tree etc. tabs and it is confusing. Is this expected? I have shared my captured trace here. Please help.

![image](https://user-images.githubusercontent.com/68080545/153760031-57445bb2-2e01-426c-bee9-51db4ea274d1.png)

![image](https://user-images.githubusercontent.com/68080545/153760086-1f5144e8-cda9-443c-aeb6-37a07f3c171e.png)

**Trace:**
[dotnet_20220212_140414.zip](https://github.com/microsoft/perfview/files/8055901/dotnet_20220212_140414.zip)

**Comment by kiranongithub** [comment]
I do not see "CPU Stacks" but only "Any Stacks". Is this expected for `dotnet-trace` related output?
![image](https://user-images.githubusercontent.com/68080545/153761686-4b5e4549-7ed7-4dfa-8191-89ed0439a747.png)

**Comment by brianrob** [comment]
No worries @kiranongithub.  Feel free to ask questions as you're learning.

What you clicked on is the "Any stacks" view, which will show a stack for any event in the trace that has a stack.  What you're seeing is the events that power the sample profiler.  These events are also used in the Thread Time views, though you won't see the event names, there.  The Any stacks view is there to allow for analysis of arbitrary events, including events that you or someone else might add, that have stacks.

**Reply by brianrob** [reply]
I should add that the reason you don't see "CPU stacks" is because `dotnet-trace` isn't capable of differentiating CPU stacks in the same way that in-kernel logging systems can.  Instead, we keep track of thread time, and do our best to do some disambiguation.  If you're looking for CPU time, the Thread Time view is going to be the closest thing that you'll get with `dotnet-trace`, and you'll just need to know that what you're looking at is all time spent on all threads.  If you really need CPU time, you will want to capture a trace using a platform logger such as ETW (PerfView can do this on Windows) or perf_events (see https://docs.microsoft.com/en-us/dotnet/core/diagnostics/trace-perfcollect-lttng).

**Reply by fuyfuy** [reply]
@brianrob thanks, that's helpful. 
Do you know if there's any resource (blog, documentation, book) that describes these and other similar details on both dotnet tools (trace, dump etc), and perfview?
It's quite critical to understand these to do performance analysis, however official documentation is very light on details.
Perfview help also doesn't describe things like "why am I seeing only Any Stacks view and no Cpu stacks".

**Reply by brianrob** [reply]
I would probably refer you to official diagnostics guidance.  @tommcdon, do you know if our existing documentation covers this?
